ci/jenkins/test.sh

```diff
@@ -274,6 +274,10 @@ function wait_for_antrea_windows_pods_ready {
         kubectl apply -f "${WORKDIR}/kube-proxy-${WINDOWS_YAML_SUFFIX}.yml"
     fi
     kubectl apply -f "${WORKDIR}/antrea-${WINDOWS_YAML_SUFFIX}.yml"
+
+    # Set trap to catch any errors from subsequent commands
+    trap collect_windows_network_info_and_logs ERR
```

Comment:
I noticed when I was troubleshooting something last week that when the Antrea Agent initialization fails on Windows, it is not easy to retrieve the logs with `kubectl logs`. I had to locate the log files from the container runtime on the Node's flesystem, and read them directly. I don't recall having the same issue on Linux.
Maybe there was some issue with my testbed... I hope that if we have the same type of failure in CI (e.g., panic in agent initialization), `collect_windows_network_info_and_logs` will be able to collect the Agent's logs successfully.